### ISSUE TITLE
Empty tracklets

### ISSUE THREAD

**Issue by GozuNaz** [DeepLabCut/DeepLabCut#1693]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

Hello,

this is my first topic, so at the beggining I wanted to thank you for your great software, documentation and community, many other threads already helped me a lot ^^,

Right now I am using maDLC 2.2.06 on fresh env and in my behaviour and I get this bug that tracklets are empty. I have two distinctable mice (one has marked tail) which can interact freely. I labeled 30 frames from 28 videos (videos are quite different) and trained my network which works quite fine (there are major ID swaps, but it is not that much work in refining tracklets). While reading documentation I noticed that I should set identity=true in config.yaml and while analising I should assemble using animal identity which really worked and I got less major ID swaps. I have done it on earlier version of maDLC (cant remember which one exactly) and then I updated DLC to 2.2.0.6 version and wanted to train the network once again with the same labled data and settings. (saving 10k, max 120k)
Prior to analysing videos evetyhing works normal and here is the problem, when I want to analyse videos with assembling using identity I get this bug that tracklets are empty (when I set it to not use animal identity while assembling it works just fine). Even when I use my network trained on earlier version of DLC I have the same issue (even tough it didint happen earlier). 
I would love to use animal identity during assemble because it reduced the number of ID swaps and here is my question is it some kind of bug, or like I read in other threads zero tracklets error means that my network is not trained well enough and I need more labled frames when animals intermingle?

Thank you,
cheers

### Operating System

Windows 10

### DeepLabCut version

2.2.0.6

### DeepLabCut mode

multi animal

### Device type

gpu

### Relevant log output

```shell
config.yaml
    # Project definitions (do not edit)
Task: Remote
scorer: Gozu
date: Feb17
multianimalproject: true
identity: true

    # Project path (change when moving around)
project_path: D:\praca\ML\Remote-Gozu-2022-02-17

    # Annotation data set configuration (and individual video cropping parameters)
video_sets:
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\EXP_KO_5.mp4:
    crop: 0, 550, 0, 380
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\EXP_KO_4.mp4:
    crop: 0, 550, 0, 380
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\EXP_KO_3.mp4:
    crop: 0, 550, 0, 380
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\EXP_KO_2.mp4:
    crop: 0, 550, 0, 380
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\EXP_KO_1.mp4:
    crop: 0, 550, 0, 380
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\CTRL_KO_6.mp4:
    crop: 0, 550, 0, 380
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\CTRL_KO_5.mp4:
    crop: 0, 550, 0, 380
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\CTRL_KO_4.mp4:
    crop: 0, 550, 0, 380
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\CTRL_KO_3.mp4:
    crop: 0, 550, 0, 380
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\CTRL_KO_1.mp4:
    crop: 0, 550, 0, 380
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\day1_wt9-05182020154959-0000.mp4:
    crop: 0, 1024, 0, 600
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\day1_wt8-05182020152929-0000.mp4:
    crop: 0, 1024, 0, 600
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\day1_wt7-05182020151130-0000.mp4:
    crop: 0, 1024, 0, 600
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\day1_wt6-05182020145325-0000.mp4:
    crop: 0, 1024, 0, 600
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\day1_wt5-05182020143434-0000.mp4:
    crop: 0, 1024, 0, 600
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\day1_ko4-05182020141654-0000.mp4:
    crop: 0, 1024, 0, 600
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\day1_ko3-05182020135627-0000.mp4:
    crop: 0, 1024, 0, 600
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\day1_ko2-05182020134006-0000.mp4:
    crop: 0, 1024, 0, 600
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\day1_ko1-05182020131911-0000.mp4:
    crop: 0, 1024, 0, 600
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\hab1_WT5.mp4:
    crop: 0, 344, 0, 224
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\hab1_WT4.mp4:
    crop: 0, 344, 0, 224
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\hab1_WT2.mp4:
    crop: 0, 344, 0, 224
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\hab1_WT1.mp4:
    crop: 0, 344, 0, 224
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\hab1_KO6.mp4:
    crop: 0, 344, 0, 224
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\hab1_KO5.mp4:
    crop: 0, 344, 0, 224
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\hab1_KO4.mp4:
    crop: 0, 344, 0, 224
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\hab1_KO3.mp4:
    crop: 0, 344, 0, 224
  D:\praca\ML\Remote-Gozu-2022-02-17\videos\hab1_KO2.mp4:
    crop: 0, 344, 0, 224
individuals:
- Dem
- Obs
uniquebodyparts: []
multianimalbodyparts:
- Ear_left
- Ear_right
- Nose
- Center
- Lateral_left
- Lateral_right
- Tail_base
- Tail_end
skeleton:
- - Center
  - Lateral_left
- - Lateral_left
  - Tail_base
- - Lateral_right
  - Tail_end
- - Ear_left
  - Nose
- - Ear_left
  - Lateral_right
- - Nose
  - Lateral_right
- - Ear_right
  - Center
- - Lateral_right
  - Tail_base
- - Center
  - Tail_base
- - Ear_left
  - Ear_right
- - Nose
  - Lateral_left
- - Ear_right
  - Nose
- - Ear_left
  - Lateral_left
- - Ear_right
  - Lateral_right
- - Tail_base
  - Tail_end
- - Lateral_left
  - Tail_end
- - Center
  - Lateral_right
- - Ear_left
  - Center
- - Ear_right
  - Lateral_left
- - Nose
  - Center
bodyparts: MULTI!
start: 0
stop: 1
numframes2pick: 20

    # Plotting configuration
skeleton_color: gray
pcutoff: 0.6
dotsize: 3
alphavalue: 0.7
colormap: rainbow

    # Training,Evaluation and Analysis configuration
TrainingFraction:
- 0.95
iteration: 0
default_net_type: dlcrnet_ms5
default_augmenter: multi-animal-imgaug
default_track_method: ellipse
snapshotindex: -1
batch_size: 8

    # Cropping Parameters (for analysis and outlier frame detection)
cropping: false
    #if cropping is true for analysis, then set the values here:
x1: 0
x2: 640
y1: 277
y2: 624

    # Refinement configuration (parameters from annotation dataset configuration also relevant in this stage)
corner2move2:
- 50
- 50
move2corner: true



Bug:

DLC network loading and video analysis starting ...
Using snapshot-120000 for model D:\praca\ML\Remote-Gozu-2022-02-17\dlc-models\iteration-0\RemoteFeb17-trainset95shuffle1
C:\Users\Tomek\anaconda3\envs\deep_lab_cut\lib\site-packages\tensorflow\python\keras\engine\base_layer_v1.py:1694: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.
  warnings.warn('`layer.apply` is deprecated and '
C:\Users\Tomek\anaconda3\envs\deep_lab_cut\lib\site-packages\tf_slim\layers\layers.py:684: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.
  outputs = layer.apply(inputs, training=is_training)
Activating extracting of PAFs
2022-02-18 15:25:57.541769: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-02-18 15:25:58.313642: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1525] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 3497 MB memory:  -> device: 0, name: NVIDIA GeForce RTX 3060 Laptop GPU, pci bus id: 0000:01:00.0, compute capability: 8.6
Starting to analyze %  D:\praca\ML\Remote-Gozu-2022-02-17\test\CTRL_KO_1.mp4
Loading  D:\praca\ML\Remote-Gozu-2022-02-17\test\CTRL_KO_1.mp4
Duration of video [s]:  540.0 , recorded with  25.0 fps!
Overall # of frames:  13500  found with (before cropping) frame dimensions:  550 380
Starting to extract posture from the video(s) with batchsize: 8
  0%|                                                                                        | 0/13500 [00:00<?, ?it/s]2022-02-18 15:26:01.917130: I tensorflow/stream_executor/cuda/cuda_dnn.cc:368] Loaded cuDNN version 8201
100%|████████████████████████████████████████████████████████████████████████████| 13500/13500 [02:46<00:00, 81.23it/s]
Video Analyzed. Saving results in D:\praca\ML\Remote-Gozu-2022-02-17\test...
Using snapshot-120000 for model D:\praca\ML\Remote-Gozu-2022-02-17\dlc-models\iteration-0\RemoteFeb17-trainset95shuffle1
Processing...  D:\praca\ML\Remote-Gozu-2022-02-17\test\CTRL_KO_1.mp4
Analyzing D:\praca\ML\Remote-Gozu-2022-02-17\test\CTRL_KO_1DLC_dlcrnetms5_RemoteFeb17shuffle1_120000.h5
13500it [00:02, 5259.09it/s]
13500it [00:00, 2685723.28it/s]
The tracklets were created (i.e., under the hood deeplabcut.convert_detections2tracklets was run). Now you can 'refine_tracklets' in the GUI, or run 'deeplabcut.stitch_tracklets'.
Processing...  D:\praca\ML\Remote-Gozu-2022-02-17\test\CTRL_KO_1.mp4
Traceback (most recent call last):
  File "C:\Users\Tomek\anaconda3\envs\deep_lab_cut\lib\site-packages\deeplabcut\gui\analyze_videos.py", line 459, in analyze_videos
    scorername = deeplabcut.analyze_videos(
  File "C:\Users\Tomek\anaconda3\envs\deep_lab_cut\lib\site-packages\deeplabcut\pose_estimation_tensorflow\predict_videos.py", line 357, in analyze_videos
    stitch_tracklets(
  File "C:\Users\Tomek\anaconda3\envs\deep_lab_cut\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py", line 1113, in stitch_tracklets
    stitcher = TrackletStitcher.from_pickle(
  File "C:\Users\Tomek\anaconda3\envs\deep_lab_cut\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py", line 504, in from_pickle
    class_ = cls.from_dict_of_dict(
  File "C:\Users\Tomek\anaconda3\envs\deep_lab_cut\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py", line 536, in from_dict_of_dict
    class_ = cls(
  File "C:\Users\Tomek\anaconda3\envs\deep_lab_cut\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py", line 457, in __init__
    raise IOError("Tracklets are empty.")
OSError: Tracklets are empty.
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by jeylau** [comment]
Hi @GozuNaz, have you tried to run `deeplabcut.create_video_with_all_detections` after video analysis? This is useful to determine whether the keypoint detections look good before continuing. Also, there should be a *_assemblies.pickle written in the folder next to your video, would you mind sharing it with us? It contains keypoint coordinates already grouped into distinct individuals, perhaps I could understand what is happening by looking at it.

**Comment by GozuNaz** [comment]
Hello @jeylau ! Thank you for your answer. So when I analyse videos and choose to assemble without using identity deeplabcut.create_video_with_all_detections this works just fine, but unfortunately when I try to assemble with usage of identity I wont even get video. 
I attache both assemblies.pickle files and also both full.pickle in case they can be helpful
[assemblies_pickle.zip](https://github.com/DeepLabCut/DeepLabCut/files/8110062/assemblies_pickle.zip)
l

**Comment by jeylau** [comment]
Hey @GozuNaz, sorry for the delay. I have found a bug where assemblies would be incorrectly discarded when using identity_only; this will be fixed with #1711.
